homeassistant/components/zha/select.py

```diff
@@ -68,731 +38,38 @@ async def async_setup_entry(
-                last_state.state.replace(" ", "_")
-            ]
-
-
-class ZHANonZCLSelectEntity(ZHAEnumSelectEntity):
-    """Representation of a ZHA select entity with no ZCL interaction."""
-
-    @property
-    def available(self) -> bool:
-        """Return entity availability."""
-        return True
-
-
-@CONFIG_DIAGNOSTIC_MATCH(cluster_handler_names=CLUSTER_HANDLER_IAS_WD)
-class ZHADefaultToneSelectEntity(ZHANonZCLSelectEntity):
-    """Representation of a ZHA default siren tone select entity."""
-
-    _unique_id_suffix = IasWd.Warning.WarningMode.__name__
-    _enum = IasWd.Warning.WarningMode
-    _attr_translation_key: str = "default_siren_tone"
-
-
-@CONFIG_DIAGNOSTIC_MATCH(cluster_handler_names=CLUSTER_HANDLER_IAS_WD)
-class ZHADefaultSirenLevelSelectEntity(ZHANonZCLSelectEntity):
-    """Representation of a ZHA default siren level select entity."""
-
-    _unique_id_suffix = IasWd.Warning.SirenLevel.__name__
-    _enum = IasWd.Warning.SirenLevel
-    _attr_translation_key: str = "default_siren_level"
-
-
-@CONFIG_DIAGNOSTIC_MATCH(cluster_handler_names=CLUSTER_HANDLER_IAS_WD)
-class ZHADefaultStrobeLevelSelectEntity(ZHANonZCLSelectEntity):
-    """Representation of a ZHA default siren strobe level select entity."""
-
-    _unique_id_suffix = IasWd.StrobeLevel.__name__
-    _enum = IasWd.StrobeLevel
-    _attr_translation_key: str = "default_strobe_level"
-
-
-@CONFIG_DIAGNOSTIC_MATCH(cluster_handler_names=CLUSTER_HANDLER_IAS_WD)
-class ZHADefaultStrobeSelectEntity(ZHANonZCLSelectEntity):
-    """Representation of a ZHA default siren strobe select entity."""
-
-    _unique_id_suffix = Strobe.__name__
-    _enum = Strobe
-    _attr_translation_key: str = "default_strobe"
-
-
-class ZCLEnumSelectEntity(ZhaEntity, SelectEntity):
-    """Representation of a ZHA ZCL enum select entity."""
-
-    _attribute_name: str
-    _attr_entity_category = EntityCategory.CONFIG
-    _enum: type[Enum]
-
-    @classmethod
-    def create_entity(
-        cls,
-        unique_id: str,
-        zha_device: ZHADevice,
-        cluster_handlers: list[ClusterHandler],
-        **kwargs: Any,
-    ) -> Self | None:
-        """Entity Factory.
-
-        Return entity if it is a supported configuration, otherwise return None
-        """
-        cluster_handler = cluster_handlers[0]
-        if ENTITY_METADATA not in kwargs and (
-            cls._attribute_name in cluster_handler.cluster.unsupported_attributes
-            or cls._attribute_name not in cluster_handler.cluster.attributes_by_name
-            or cluster_handler.cluster.get(cls._attribute_name) is None
-        ):
-            _LOGGER.debug(
-                "%s is not supported - skipping %s entity creation",
-                cls._attribute_name,
-                cls.__name__,
+    def restore_external_state_attributes(self, state: State) -> None:
+        """Restore entity state."""
+        if state.state and state.state != STATE_UNKNOWN:
```

Comment:
How about unavailable? Do we want to restore that?

Reply:
I'm wondering, why is the state restored for the select entities here, but not seeing this pattern in other entities?

Reply:
Most select entities directly change attributes on Zigbee devices, which are reflected in the zigpy attribute cache (db).
However, siren entities have non-ZCL select configuration entities that are "fake", i.e. not coupled to attributes on a Zigbee device. ZHA uses their state when a siren is turned on to send the configuration parameters then.
But since zigpy doesn't store state then and ZHA can't (yet), those are restored from HA.

Reply:
Ah, they're split in ZHA. It's not used for "normal select entities" there.
- [normal select entities backed by ZCL cache](https://github.com/zigpy/zha/blob/378e53641d02a450a99a67dccf79e97324196d77/zha/application/platforms/select.py#L266-L272) (empty `restore_external_state_attributes`)
- [non ZCL select entities for sirens](https://github.com/zigpy/zha/blob/378e53641d02a450a99a67dccf79e97324196d77/zha/application/platforms/select.py#L112-L119) (`restore_external_state_attributes` used here only)